build/scripts/build.ps1

```diff
@@ -435,6 +435,9 @@ function Test-XUnitCoreClr() {
 # Core function for running our unit / integration tests tests
 function Test-XUnit() { 
 
+    # Used by tests to locate dotnet CLI
+    $env:DOTNET_INSTALL_DIR = Split-Path $dotnet    
```

Comment:
>$env:DOTNET_INSTALL_DIR  [](start = 4, length = 24)

This approach overall seems fragile. Our tests run in many contexts, most of which are not driven by this script. Hence there is no way to depend on this value being set.

Reply:
A better idea? The test runner tries to use the `DOTNET_INSTALL_DIR` and falls back to looking for the right version of the SDK on `PATH`.

Reply:
Unfortunately I don't have a better idea. 

---
In reply to: [176506227](https://github.com/dotnet/roslyn/pull/25325#discussion_r176506227) [](ancestors = 176506227)